src/riak_kv_pb_timeseries.erl

```diff
@@ -362,6 +374,41 @@ row_to_key(Row, DDL, Mod) ->
     riak_kv_ts_util:encode_typeval_key(
       riak_ql_ddl:get_partition_key(DDL, Row, Mod)).
 
+%%%%%%%%
+%% Utility functions for batch delivery of records
+termsize(Term) ->
+    size(term_to_binary(Term)).
+
+pick_batch_option(_, _, Records, _, 1) ->
+    {individual, Records};
+pick_batch_option(true, MaxBatch, Records, SampleSize, _NumRecs) ->
+    {batches, create_batches(Records,
+                             estimated_row_count(SampleSize, MaxBatch))};
+pick_batch_option(false, _, Records, _, _) ->
+    {individual, Records}.
+
+estimated_row_count(SampleRowSize, MaxBatchSize) ->
+    %% Assume some rows will be larger, so introduce a fudge factor of
+    %% roughly 10 percent.
+    RowSizeFudged = (SampleRowSize * 10) div 9,
+    MaxBatchSize div RowSizeFudged.
+
+create_batches(Rows, MaxSize) when length(Rows) =< MaxSize ->
+    [Rows];
+%% May be a more efficient way to do this. Take a list of arbitrary
+%% data (expected to be a list of lists for this use case) and create
+%% a list of MaxSize lists.
+create_batches(Rows, MaxSize) ->
+    create_batches(Rows, MaxSize, MaxSize, [], []).
+
+create_batches([], _Counter, _Max, ThisBatch, AllBatches) ->
+    AllBatches ++ [ThisBatch];
```

Comment:
Does the order of batch groups matter here? I wonder if you could reassemble list elements with a little more efficient `[ThisElement|Rest]`.